ArcGISRuntimeSDKQt_CppSamples/Layers/DisplayDimensions/DisplayDimensions.cpp

```diff
@@ -0,0 +1,190 @@
+// [Legal]
+// Copyright 2021 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#ifdef PCH_BUILD
+#include "pch.hpp"
+#endif // PCH_BUILD
+
+#include "DisplayDimensions.h"
+
+#include "DimensionLayer.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "MobileMapPackage.h"
+
+#include <QDir>
+
+using namespace Esri::ArcGISRuntime;
+
+namespace
+{
+  // Helper method to get cross platform data path.
+  QString defaultDataPath()
+  {
+    QString dataPath;
+
+    #ifdef Q_OS_ANDROID
+      dataPath = "/sdcard";
+    #elif defined Q_OS_IOS
+      dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+    #else
+      dataPath = QDir::homePath();
+    #endif
+
+    return dataPath;
+  }
+
+  // Sample MMPK location.
+  const QString edinburghPylonFilePath {"/ArcGIS/Runtime/Data/mmpk/Edinburgh_Pylon_Dimensions.mmpk"};
+}
+
+DisplayDimensions::DisplayDimensions(QObject* parent /* = nullptr */):
+  QObject(parent)
+{
+  QString mapPackagePath = defaultDataPath() + edinburghPylonFilePath;
+  m_mmpk = new MobileMapPackage(mapPackagePath, this);
+
+  // Make connections between the mmpk's doneLoading and errorOccurred signal and the addMapToMapView and handleError methods respectively.
+  connect(m_mmpk, &MobileMapPackage::doneLoading, this, &DisplayDimensions::addMapToMapView);
+  connect(m_mmpk, &MobileMapPackage::errorOccurred, this, &DisplayDimensions::handleError);
+
+  // Load the mmpk.
+  m_mmpk->load();
+}
+
+DisplayDimensions::~DisplayDimensions() = default;
+
+void DisplayDimensions::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<DisplayDimensions>("Esri.Samples", 1, 0, "DisplayDimensionsSample");
+}
+
+void DisplayDimensions::addMapToMapView(const Error& error)
+{
+  // If no error's have occurred, the mmpk is loaded, and there is only one map in the mmpk, the map in the mmpk can be
+  // assigned to the MapView and the viewpoint set. Any errors will be managed by the handleError method.
+  if (error.isEmpty() && m_mmpk->loadStatus() == LoadStatus::Loaded && m_mmpk->maps().count() > 0)
+  {
+    // Enable the checkboxes.
+    setCheckBoxesEnabled(true);
+
+    // Assign the map in the mmpk to m_map.
+    m_map = m_mmpk->maps().at(0);
+    // Set the viewpoint.
+    m_map->setInitialViewpoint(Viewpoint(Point(-3.3098678, 55.9074044, SpatialReference::wgs84()), 30000));
```

Comment:
i don't think you should need to do this: the mmpk can encode a starting location if you want one for the sample...